### PR TITLE
refactor(ses): move template rendering and the ObjectMapper into SesTemplateService

### PR DESCRIPTION
**Pull request by okinaka**
## Summary

Follow-up to #2356: with the store split complete (#2881, #2984), this moves the remaining
template-domain logic out of the `SesService` facade into `SesTemplateService`: `TestRenderTemplate`
(data parsing, variable substitution, and the MIME assembly with its helpers), plus the
`applyTemplateData` substitution shared with the templated-send paths, which the facade now calls
on the service. The service takes `ObjectMapper` and a constructor-passed `SecureRandom` (for the
MIME boundary) as dependencies.

Two facade-side cleanups fall out of this: the `objectMapper` field and constructor parameter are
removed (rendering was its only remaining use), and the static `BOUNDARY_RANDOM` disappears, so the
`--initialize-at-run-time` entry for `SesService` in `application.yml` is removed as well (the
constructor-passed instance needs no native-image registration; the guard test confirms).

No behavior change: request/response shapes, error messages, and rendering semantics are untouched.

## Type of change

- [ ] Bug fix (`fix:`)
- [ ] New feature (`feat:`)
- [ ] Breaking change (`feat!:` or `fix!:`)
- [x] Docs / chore

## AWS Compatibility

Pure code motion behind the facade; no wire-visible change. The full SES test suite (1037 tests)
plus the native-image runtime-initialization guard passes unchanged, pinning the existing rendering
behavior (variable substitution edge cases, MIME structure, transfer-encoding selection, and the
`InvalidRenderingParameter` / `MissingRenderingAttribute` wordings).

## Checklist

- [x] `./mvnw test` passes locally
- [x] New or updated integration test added (not applicable: behavior-preserving refactor; the existing rendering unit tests now target the moved statics directly)
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

### CODE REVIEW

File: src/main/java/io/github/hectorvent/floci/services/ses/SesTemplateService.java [hunk 1]
```diff
@@ -32,16 +46,27 @@ public class SesTemplateService {
...
-        this.templateStore = storageFactory.create("ses", "ses-templates.json",
-                new TypeReference<Map<String, EmailTemplate>>() {});
+    public SesTemplateService(StorageFactory storageFactory, ObjectMapper objectMapper) {
+        this(storageFactory.create("ses", "ses-templates.json",
+                new TypeReference<Map<String, EmailTemplate>>() {}), objectMapper, new SecureRandom());
     }
 
     SesTemplateService(StorageBackend<String, EmailTemplate> templateStore) {
+        this(templateStore, new ObjectMapper(), new SecureRandom());
+    }
```
[the remark this answers is not in the document]
Comment (okinaka) [reply src/main/java/io/github/hectorvent/floci/services/ses/SesTemplateService.java]: Fixed in dbd30617: the convenience constructor is removed, so every non-CDI caller now supplies the ObjectMapper (and SecureRandom) explicitly, and the production path keeps receiving the JacksonConfig-customized mapper via injection. The CRUD unit test passes a plain mapper deliberately: none of its assertions parse JSON, and the rendering tests that do exercise parsing already share one explicit MAPPER instance.

**Review by pgermosen — APPROVED** [review]
Didn't just take the no-behavior-change claim at face value, diffed the moved method bodies directly and they're byte-for-byte identical, the only behavioral-adjacent change is the boundary randomness moving from a static field to constructor-injected, which is exactly the stated native-image goal, not a quality change. The ObjectMapper finding is genuinely fixed too, that convenience constructor is gone and the one remaining plain-mapper test has no rendering assertions to be affected by it.

Nothing outstanding.
